orttraining/orttraining/training_api/include/onnxruntime_training_cxx_api.h

```diff
@@ -0,0 +1,151 @@
+                  const std::optional<std::basic_string<ORTCHAR_T>>& optimizer_model_path =
+                      std::optional<std::basic_string<ORTCHAR_T>>());
+
+  /** \brief Run the train step returning results in an Ort allocated vector.
+   *
+   * Wraps OrtTrainingApi::TrainStep
+   *
+   * \param[in] input_values Array of Value objects in the order expected by the training model.
+   * \return A std::vector of Value objects that represents the output of the forward pass.
+   */
+  std::vector<Value> TrainStep(const std::vector<Value>& input_values);
+
+  /** \brief Lazily resets the gradients of the trainable parameters.
+   *
+   * Wraps OrtTrainingApi::ResetGrad
+   *
+   */
+  void ResetGrad();
+
+  /** \brief Run the evaluation step returning results in an Ort allocated vector.
+   *
+   * Wraps OrtTrainingApi::EvalStep
+   *
+   * \param[in] input_values Array of Value objects in the order expected by the eval model.
+   * \return A std::vector of Value objects that represents the output of the eval pass.
+   */
+  std::vector<Value> EvalStep(const std::vector<Value>& input_values);
+
+  /** \brief Set the learning rate to be used by the optimizer for parameter updates.
+   *
+   * Wraps OrtTrainingApi::SetLearningRate
+   *
+   * \param[in] learning_rate float value representing the constant learning rate to be used.
+   */
+  void SetLearningRate(float learning_rate);
+
+  /** \brief Get the current learning rate that is being used by the optimizer.
+   *
+   * Wraps OrtTrainingApi::GetLearningRate
+   *
+   * \return float representing the current learning rate.
+   */
+  float GetLearningRate();
+
+  /** \brief Register the linear learning rate scheduler for the training session.
+   *
+   * Wraps OrtTrainingApi::RegisterLinearLRScheduler
+   *
+   * \param[in] warmup_step_count Number of steps in the warmup phase.
+   * \param[in] total_step_count Total number of training steps.
+   * \param[in] initial_lr Initial learning rate to use.
+   */
+  void RegisterLinearLRScheduler(const int64_t warmup_step_count, const int64_t total_step_count,
+                                 const float initial_lr);
+
+  /** \brief Updates the learning rate based on the lr scheduler.
+   *
+   * Wraps OrtTrainingApi::SchedulerStep
+   *
+   */
+  void SchedulerStep();
+
+  /** \brief Runs the optimizer model and updates the model parameters.
+   *
+   * Wraps OrtTrainingApi::OptimizerStep
+   *
+   */
+  void OptimizerStep();
+};
+
+/** \brief Loads the checkpoint at provided path and returns the checkpoint state
+ *
+ * Wraps OrtTrainingApi::LoadCheckpoint
+ *
+ */
+CheckpointState LoadCheckpoint(const std::basic_string<ORTCHAR_T>& path_to_checkpoint);
+
+/** \brief Saves the state of the training session to a checkpoint file provided by the given path.
+ *
+ * Wraps OrtTrainingApi::SaveCheckpoint
+ *
+ */
+void SaveCheckpoint(const TrainingSession& session, const std::basic_string<ORTCHAR_T>& path_to_checkpoint,
```

Comment:
I think the `SaveCheckpoint` should be updated to take in a checkpoint_state and save it to a path. It should not be coupled with `TrainingSession` since this does not allow users to add meaningful information (checkpoint properties) to the checkpoint being saved such as saving the epoch, best score.

But I was going to get this in another pull request.